sklearn/neighbors/base.py

```diff
@@ -258,6 +258,19 @@ def _fit(self, X):
         else:
             raise ValueError("algorithm = '%s' not recognized"
                              % self.algorithm)
+
+        if self.n_neighbors is not None:
```

Comment:
It would be possible to add a check in "predict", too, in a place where ``None`` is not valid, but I'm not sure that is necessary.